app/models/language.rb

```diff
@@ -1,2 +1,30 @@
 class Language < ActiveRecord::Base
+
+  # create_table :languages do |t|
+  #   t.string :name
+  #   t.string :code
+  #
+  #   t.timestamps
+
+
+  # returns language_id for given code
+  def self.resolve_id(code = nil)
+    code ||= I18n.language_code
+    Language.where({code: code.upcase}).first.try(:id)  #todo: use a cache to optimize
+  end
+
+  #todo: use a cache here also
```

Comment:
i didn't know that term. but i use that technique for single value caching.  in this case, i can memoize a cached map as a class var, which is then used to perform a local lookup. 
i still prefer to defer these optimization until things have stabilized a bit.